graphql_compiler/schema_transformation/merge_schemas.py

```diff
@@ -248,3 +308,274 @@ def _process_generic_type_definition(generic_type, schema_id, existing_scalars,
+
+
+def _check_field_reference_is_valid(type_name_to_definition, name_to_schema_id, field_reference):
+    """Check that the field reference refers to a valid field.
+
+    In particular, check that the field reference is on a type that exists in the correct
+    schema, and that the type contains the field of the expected name.
+
+    Args:
+        type_name_to_definition: Dict[str, (Interface/Object/Union)TypeDefinition], mapping
+                                 name of types to their definitions
+        name_to_schema_id: Dict[str, str], mapping type name to id of the schema that the
+                           type is from
+        field_reference: FieldReference, what we check the validity of
+
+    Raises:
+        - InvalidCrossSchemaEdgeError if the field reference refers nonexistent schemas,
+          types, or fields
+    """
+    schema_id = field_reference.schema_id
+    type_name = field_reference.type_name
+    field_name = field_reference.field_name
+
+    # Error if the type is nonexistent (includes if type is an enum or scalar)
+    # TODO: Consider improving error message if type is an enum or scalar
+    if type_name not in type_name_to_definition:
+        raise InvalidCrossSchemaEdgeError(
+            u'Type "{}" specified in the field reference "{}" is not found '
+            u'in the merged schema.'.format(type_name, field_reference)
+        )
+
+    # Error if the type is in a wrong or nonexistent schema
+    if name_to_schema_id[type_name] != schema_id:
+        raise InvalidCrossSchemaEdgeError(
+            u'Type "{}" specified in the field reference "{}" is expected to be in '
+            u'schema "{}", but is instead bound in schema "{}"'.format(
+                type_name, field_reference, schema_id,
+                name_to_schema_id[type_name]
+            )
+        )
+
+    # Error if the type doesn't have the expected field
+    type_definition = type_name_to_definition[type_name]
+    type_fields = type_definition.fields
+    if not any(field.name.value==field_name for field in type_fields):
+        raise InvalidCrossSchemaEdgeError(
+            u'Field "{}" is not found under type "{}" in schema "{}", as expected by the '
+            u'field reference "{}".'.format(
+                field_name, type_name, schema_id, field_reference
+            )
+        )
+
+
+def _check_field_types_are_matching_scalars(type_name_to_definition, scalars, cross_schema_edge):
+    """Check that stitched fields in a cross schema edge are of the same scalar type.
+
+    It is also legal for fields to be of a NonNull wrapped scalar type.
+
+    Args:
+        type_name_to_definition: Dict[str, (Interface/Object/Union)TypeDefinition], mapping
+                                 name of types to their definitions
+        scalars: Set[str], names of all scalars in the merged_schema so far
+        cross_schema_edge: CrossSchemaEdge, the edge that we check the validity of
```

Comment:
nit CrossSchemaEdge namedtuple. When you mention that a type is a namedtuple, you give the reader an intuition of how to interact with it.